### PR TITLE
#9386: copy rustfmt.toml to node-template

### LINKED ISSUE #9386: Include the `rustfmt.toml` when generating the `node-template`
With #8982 and #9381, the `node-template` now will be formatted with the same rustfmt settings, and we would like to propagate this file in the [the template generation tool](https://github.com/paritytech/substrate/blob/master/.maintain/node-template-release/src/main.rs) so that the [devhub node template](https://github.com/substrate-developer-hub/substrate-node-template) will always include the same `rustfmt.toml` file.

### PR DESCRIPTION
**Pull request by ericjmiller**
Fixes #9386
Description
The[ node template generator](https://github.com/paritytech/substrate/blob/master/.maintain/node-template-release/src/main.rs) now adds the `rustfmt.toml` from root level.

Oh, and I couldn't resist running `rustfmt` on this file.

### CODE REVIEW

File: .maintain/node-template-release/src/main.rs [hunk 1]
```diff
@@ -233,20 +250,30 @@ fn main() {
...
 			update_top_level_cargo_toml(&mut cargo_toml, workspace_members, &node_template_path);
 		}
 
 		write_cargo_toml(&t, cargo_toml);
 	});
 
+	// adding root rustfmt to node template build path
+	let node_template_rustfmt_toml_path = node_template_path.join("rustfmt.toml");
+	let root_rustfmt_toml =
+		&options.node_template.parent().unwrap().parent().unwrap().join("rustfmt.toml");
```
Comment (bkchr) [inline .maintain/node-template-release/src/main.rs:265]:
```suggestion
		&options.node_template.join("../rustfmt.toml");
```

Doesn't this work?

**Review by bkchr — APPROVED** [review]

File: .maintain/node-template-release/src/main.rs (hunk 1 above)
Comment (ericjmiller) [reply .maintain/node-template-release/src/main.rs:265]: I don't think so.  Here's the command that calls this:
```bash
cargo run $PROJECT_ROOT/bin/node-template $PROJECT_ROOT/$PATH_TO_ARCHIVE
```
You'll need to go up two levels from the `options.node_template` arg.  Also, explicity usage of `...` wouldn't be as explicity in error handling if the directory ever moves.  That's okay if it's more legible, but it was at least need to be:
```rust
&options.node_template.join("../../rustfmt.toml");
```
Does that make sense or am I just completely confused here?

Comment (ericjmiller) [reply .maintain/node-template-release/src/main.rs:265]: @danforbes ^^^

Comment (bkchr) [reply .maintain/node-template-release/src/main.rs:265]: Sorry, I meant `../../`. I updated the suggestion above!

**Review by TriplEight — COMMENTED** [review]
Please resolve the remaining review and it's good to go!

**Review by TriplEight — APPROVED** [review]
Thanks! Now you just need to rebase on master to make that job pass.

**Pull request merged**